File: include/PolyWorksSDK/samples/applications/IMAlign/VisualBasic.Net/VBNetSimpleAlignment/MainForm.vb
Public Class MainForm

    ' Global COM Message Filter to prevent COM calls from being rejected if the server application is temporarily busy
    Private comFilter As CComMessageFilter = Nothing

    ' Upon loading the Form
    Private Sub MainForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        comFilter = New CComMessageFilter

    End Sub

    ' Cleanup upon destruction of the form
    Private Sub MainForm_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed

        comFilter = Nothing

    End Sub

    ' Main action of this application
    Private Sub ButtonStart_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ButtonStart.Click

        Dim imAlignObject As New IMAlignLib.IMAlign

        TextLog.Text = ""

        ' If the server does not exists, return
        If imAlignObject Is Nothing Then

            Exit Sub

        End If

        ' Obtain IMAlign's current project
        Dim projectObject As IMAlignLib.IIMAlignProject = Nothing
        imAlignObject.ProjectGetCurrent(projectObject)

        If projectObject Is Nothing Then

            System.Runtime.InteropServices.Marshal.ReleaseComObject(imAlignObject)
            imAlignObject = Nothing
            Exit Sub

        End If

        ' Obtain a Command Center from the current IMAlign project
        Dim commandCenterObject As IMAlignLib.IIMCommandCenter
        projectObject.CommandCenterCreate(commandCenterObject)

        ' Send commands
        Dim buffer As String

        If Not commandCenterObject Is Nothing Then

            buffer = "FILE NEW_PROJECT"
            TextLog.Text = TextLog.Text & buffer & vbCrLf
            commandCenterObject.CommandExecute(buffer)

            buffer = "CONFIG DIGITIZERS NAME ( ""Generic Close Range (in mm)"" )"
            TextLog.Text = TextLog.Text & buffer & vbCrLf
            commandCenterObject.CommandExecute(buffer)

            buffer = "FILE IMPORT_IMAGE 3D_DIGITIZED_DATASET PLANAR_GRID DIRECTORY ( """ & TextSource.Text & """, , ""PIF"" )"
            TextLog.Text = TextLog.Text & buffer & vbCrLf
            commandCenterObject.CommandExecute(buffer)

            buffer = "ALIGN START(0.000001, 10)"
            TextLog.Text = TextLog.Text & buffer & vbCrLf
            commandCenterObject.CommandExecute(buffer)

        End If

        ' Cleanup
        System.Runtime.InteropServices.Marshal.ReleaseComObject(commandCenterObject)
        System.Runtime.InteropServices.Marshal.ReleaseComObject(projectObject)
        System.Runtime.InteropServices.Marshal.ReleaseComObject(imAlignObject)

        commandCenterObject = Nothing
        projectObject = Nothing
        imAlignObject = Nothing

    End Sub

End Class

File: include/PolyWorksSDK/samples/applications/IMAlign/CSharp/CSharpSimpleAlignment/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CSharpSimpleAlignment
{
    public partial class MainForm : Form
    {
        // =========================================================================================
        // ========================================= Methods =======================================

        // =========================================================================================
        // Constructor
        // -----------------------------------------------------------------------------------------
        public MainForm()
        {
            InitializeComponent();
        }

        // =========================================================================================
        // Main action of this application
        // -----------------------------------------------------------------------------------------
        private void ButtonStart_Click(object sender, EventArgs e)
        {
            IMAlignLib.IMAlign imAlignObject = new IMAlignLib.IMAlign();

            TextLog.Text = "";

            // If the server does not exists, return
            if ( imAlignObject == null )
            {
                return;
            }

            // Obtain IMAlign's current project
            IMAlignLib.IIMAlignProject imAlignProject = null;
            imAlignObject.ProjectGetCurrent( out imAlignProject );

            if ( imAlignProject == null )
            {
                imAlignObject = null;
                return;
            }

            // Obtain a Command Center from the current IMAlign project
            IMAlignLib.IIMCommandCenter imCommandCenter = null;
            imAlignProject.CommandCenterCreate( out imCommandCenter );

            // Send commands
            String buffer;

            if ( imCommandCenter != null )
            {
                buffer = "FILE NEW_PROJECT";
                TextLog.Text = String.Concat (TextLog.Text, buffer, "\r\n" );
                imCommandCenter.CommandExecute( buffer );

                buffer = "CONFIG DIGITIZERS NAME ( \"Generic Close Range (in mm)\" )";
                TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                imCommandCenter.CommandExecute( buffer );

                buffer = String.Concat( "FILE IMPORT_IMAGE 3D_DIGITIZED_DATASET PLANAR_GRID DIRECTORY ( \"", TextSource.Text, "\", , \"PIF\" )" );
                TextLog.Text = String.Concat( TextLog.Text, buffer , "\r\n" );
                imCommandCenter.CommandExecute( buffer );

                buffer = "ALIGN START(0.000001, 10)";
                TextLog.Text = String.Concat( TextLog.Text, buffer , "\r\n" );
                imCommandCenter.CommandExecute( buffer );
            }

            // Cleanup
            System.Runtime.InteropServices.Marshal.ReleaseComObject( imCommandCenter );
            System.Runtime.InteropServices.Marshal.ReleaseComObject( imAlignProject );
            System.Runtime.InteropServices.Marshal.ReleaseComObject( imAlignObject );
            imCommandCenter = null;
            imAlignProject = null;
            imAlignObject = null;
        }
    }
}